app/src/main/java/org/oppia/app/player/exploration/ExplorationActivityPresenter.kt

```diff
@@ -36,4 +44,17 @@ class ExplorationActivityPresenter @Inject constructor(private val activity: App
       R.id.exploration_fragment_placeholder
     ) as ExplorationFragment?
   }
+
+  fun stopExploration() {
+    explorationDataController.stopPlayingExploration().observe(activity, Observer<AsyncResult<Any?>> {
+      when {
+        it.isPending() -> logger.d("ExplorationActivity", "Stopping exploration")
+        it.isFailure() -> logger.e("ExplorationActivity", "Failed to stop exploration", it.getErrorOrNull()!!)
+        else -> {
+          logger.d("ExplorationActivity", "Successfully stopped exploration")
+          (activity as ExplorationActivity).routeBackToTopicActivity()
```

Comment:
Could just finish() here to simplify.

Reply:
Done.
Changed to 
`(activity as ExplorationActivity).finish()`